### DISCUSSION TITLE
artillery@2.0.0-37 throwing an error - No matching version found for aws-sdk@^2.1338.0

### DISCUSSION THREAD

**Discussion by anildel8s** [artilleryio/artillery#2165] [Artillery]
artillery@2.0.0-37 throwing an error - No matching version found for aws-sdk@^2.1338.0

 ERR_PNPM_NO_MATCHING_VERSION  No matching version found for aws-sdk@^2.1338.0

This error happened while installing the dependencies of artillery@2.0.0-37

The latest release of aws-sdk is "2.1226.0".

I have tried via npm also but got the same issue. 
npm verb stack aws-sdk: No matching version found for aws-sdk@^2.1338.0.
npm verb stack     at module.exports (C:\DEV\nodejs-18.4.0\node_modules\npm\node_modules\npm-pick-manifest\lib\index.js:209:23)
npm verb stack     at RegistryFetcher.manifest (C:\DEV\nodejs-18.4.0\node_modules\npm\node_modules\pacote\lib\registry.js:125:22)
npm verb stack     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
npm verb stack     at async Promise.all (index 15)
npm verb stack     at async [buildDepStep] (C:\DEV\nodejs-18.4.0\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.js:1093:5)
npm verb stack     at async Arborist.buildIdealTree (C:\DEV\nodejs-18.4.0\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\build-ideal-tree.j
s:218:7)
npm verb stack     at async Promise.all (index 1)
npm verb stack     at async Arborist.reify (C:\DEV\nodejs-18.4.0\node_modules\npm\node_modules\@npmcli\arborist\lib\arborist\reify.js:153:5)
npm verb stack     at async Install.exec (C:\DEV\nodejs-18.4.0\node_modules\npm\lib\commands\install.js:156:5)
npm verb stack     at async module.exports (C:\DEV\nodejs-18.4.0\node_modules\npm\lib\cli.js:78:5)
npm verb cwd C:\DEV\localrepos\artillery-poc
npm verb Windows_NT 10.0.19045
npm verb node v18.4.0
npm verb npm  v8.12.1
npm ERR! code ETARGET
npm ERR! notarget No matching version found for aws-sdk@^2.1338.0.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm verb exit 1
npm timing npm Completed in 48675ms
npm verb unfinished npm timer reify 1695661265867
npm verb unfinished npm timer reify:loadTrees 1695661265905
npm verb code 1

**Comment by bernardobridge** [comment]
Hello @anildel8s 👋 ,

I cannot replicate this. If you check https://www.npmjs.com/package/aws-sdk?activeTab=versions, you'll see versions above and including 2.1338.0 exist. Are you using a private registry like Artifactory, Verdaccio, or similar? 

If not, then this is possibly an issue with your npm or pnpm installation or cache, and might be running into the issues described [here](https://stackoverflow.com/questions/44331005/npm-error-no-matching-version-found-for).

**Reply by anildel8s** [reply]
hi, thanks for the quick response.  I am using nexus repository manager.  Are there any specific things to check when using private registry?
I will checkout the link you have provided meanwhile and see if any of that solves the issue for me by repointing the registry to npmjs.

**Reply by bernardobridge** [reply]
Hi @anildel8s . There shouldn't be additional config needed if you're just running `npm install`. However, there's a chance that because you're using a private registry, version `aws-sdk@^2.1338.0` doesn't exist in your private registry, and it's not pulling it from npm if it doesn't exist in your private registry? I'm just guessing here.

What I meant with that link is that it definitely exists on the public NPM registry, which is why I don't get that error. So if you are able to point to the npmjs registry, it should work.

**Comment by anildel8s** [comment]
Worked for me after re-pointing to npmjs registry.
